lua/ide/panels/panel.lua

```diff
@@ -274,6 +276,28 @@ Panel.new = function(tab, position, components)
         for _, f in ipairs(restores) do
             f()
         end
+
+        if not self.has_opened_once then
```

Comment:
Would you mind breaking this up a bit into local functions to avoid the "end" waterfall here?

Reply:
Sure, I can do that shortly.